src/components/NanoContract/NanoContractDetails.component.js

```diff
@@ -0,0 +1,211 @@
+
+  const onAddressChange = (newAddress) => {
+    changeNcAddress(newAddress);
+    dispatch(nanoContractAddressChangeRequest({ newAddress, ncId: nc.ncId }));
+  }
+
+  const navigatesToNanoContractTransaction = (tx) => {
+    navigation.navigate('NanoContractTransaction', { tx });
+  };
+
+  useEffect(() => {
+    if (txHistory.length === 0) {
+      /* The first time we load the Nano Contract details its transaction history is empty.
+       * The second time it is garanteed that its transaction history is not empty,
+       * because a Nano Contract must have at least the 'initialize' transaction to exists.
+       * For the first transaction history load we don't need to specify the `after` param,
+       * it will be set during the load.
+       */
+      dispatch(nanoContractHistoryRequest({ ncId: nc.ncId, after: null }));
+    }
+  }, []);
+
+  const handleMoreTransactions = () => {
+    console.log('fetching more. After', after);
+    if (after == null) {
+      /* This situation is unlikely to happen because on the first transactions history load
+       * the `after` is assigned with the hash of the last transaction in the list.
+       */
+      return;
+    }
+    dispatch(nanoContractHistoryRequest({ ncId: nc.ncId, after }));
+  };
+
+  /* If an error happens on loading transactions history, an error feedback
+   * content must be presented to the user, and must have precedence over
+   * all other content.
+   */
+  const hasError = () => error != null;
+  const isEmpty = () => txHistory.length === 0 && !hasError();
+  const notEmpty = () => !isEmpty() && !hasError();
+
+  return (
+    <Wrapper>
+      <NanoContractDetailsHeader
+        nc={nc}
+        address={ncAddress}
+        onAddressChange={onAddressChange}
+      />
+      {isLoading && (
+        <Spinner style={styles.center} size={14} animating />
+      )}
+      {hasError()
+        && (<ErrorLoadingTransaction ncId={nc.ncId} error={error} />)}
+      {isEmpty()
+        && (<NoNanoContractTransaction />)}
+      {notEmpty()
+        && (
+          <HathorFlatList
+            data={txHistory}
+            renderItem={({ item }) => (
+              <NanoContractTransactionsListItem
+                item={item}
+                onPress={() => navigatesToNanoContractTransaction(item)}
+              />
+            )}
+            keyExtractor={(item) => item.txId}
+            refreshing={isLoading}
+            onRefresh={handleMoreTransactions}
+          />
+        )}
+    </Wrapper>
+  );
+};
+
+/**
+ * @param {Object} props
+ * @param {Object?} props.children Either a react component or a react element
+ */
+const Wrapper = ({ children }) => (
+  <View style={styles.wrapper}>
+    {children}
+  </View>
+);
+
+/**
+ * Renders a feedback for the lack of transactions.
+ */
+export const NoNanoContractTransaction = () => (
+  <FeedbackContent
+    title={t`No Nano Contract Transactions`}
+    message={t`There is not transaction to show yet.`}
```

Comment:
The only possibility for no transaction is when it's loading the history, so I think the best would be to have a `Loading nano contract history` message here. What do you think?

Reply:
Yeah! I've added a loading feedback, but yeah, this message seems to be better.